7b.UEFI_Tool_Kit.md/README.md

```diff
@@ -0,0 +1,453 @@
+        Sec                                           EFI_SECTION_FIRMWARE_VOLUME_IMAGE       0xe0004
+         FV     8C8CE578-8A3D-4F1C-9935-896185C32DD3                                          0xe0000
+          File  1B45CC0A-156A-428A-AF62-49864DA0E6E6  EFI_FV_FILETYPE_FREEFORM                   0x2c
+           Sec                                        EFI_SECTION_RAW                            0x14
+          File  FFFFFFFF-FFFF-FFFF-FFFF-FFFFFFFFFFFF  EFI_FV_FILETYPE_FFS_PAD                    0x40
+          File  52C05B14-0B98-496C-BC3B-04B50211D680  EFI_FV_FILETYPE_PEI_CORE                 0xc4fa
+           Sec                                        EFI_SECTION_RAW                            0x3c
+           Sec                                        EFI_SECTION_PE32                         0xc484
+           Sec  PeiCore                               EFI_SECTION_USER_INTERFACE                 0x14
+           Sec                                        EFI_SECTION_VERSION                         0xe
+    ...
+
+This format is compact and easy for humans reading, but not ideal for machine
+consumption. Use the `json` command to print everything (including much more
+metadata) as JSON:
+
+    $ utk OVMF.rom json | less
+
+Combine `utk` with the JSON query command, `jq` (`sudo apt-get install jq`),
+and other UNIX commands to quickly write powerful queries. For example, the
+following lists all the GUIDs, sorted and without duplicates:
+
+    $ utk OVMF.rom json | jq -r '..|.GUID?|select(type=="string")' | sort -u
+    00000000-0000-0000-0000-000000000000
+    0167CCC4-D0F7-4F21-A3EF-9E64B7CDCE8B
+    0170F60C-1D40-4651-956D-F0BD9879D527
+    021722D8-522B-4079-852A-FE44C2C13F49
+    025BBFC7-E6A9-4B8B-82AD-6815A1AEAF4A
+    ...
+
+To only print the JSON for specific files, use the find command:
+
+    # The find command uses a regex to match on the name or GUID.
+    # These three examples find and print the JSON for the same file:
+    $ utk OVMF.rom find 'Sh.*'
+    $ utk OVMF.rom find 'Shell'
+    $ utk OVMF.rom find 7C04A583-9E3E-4F1C-AD65-E05268D0B4D1
+    {
+            "Header": {
+                    "UUID": {
+                            "UUID": "7C04A583-9E3E-4F1C-AD65-E05268D0B4D1"
+                    },
+                    "Type": 9,
+                    "Attributes": 0
+            },
+            "Type": "EFI_FV_FILETYPE_APPLICATION",
+            "Sections": [
+                    ...
+            ],
+            "ExtractPath": "",
+            "DataOffset": 24
+    }
+
+Note that UEFI uses GUIDs to identify files. Some files also have a name which
+is stored within the file's UI section. Like `find`, most of UTKs commands let
+you match a file by its name or GUID.
+
+The examples up until now have only dealt with file metadata and not the file's
+contents. The `extract <DIR>` command saves the file's contents to a file
```

Comment:
This sentence is a little confusing, it seems to imply that extract works on a single UEFI file. Instead extract extracts an entire UEFI firmware image, which ends up creating a tree of extracted files